scripts/db-heal-test.py

```diff
@@ -0,0 +1,238 @@
+parser.add_argument("--cp-granularity", type=int,
+                    help="How often will Carmen create checkpoints (in blocks).", default=10)
+
+args = parser.parse_args()
+
+aida_path = args.aida
+aida_db_path = args.aida_db
+tmp_path = args.tmp
+number_of_iterations = args.iter
+window = args.window
+checkpoint_granularity = args.cp_granularity
+
+# Mark first checkpoint
+latest_checkpoint = checkpoint_granularity
+
+if aida_path == "":
+    print("please set Aida using --aida")
+    exit(1)
+if aida_db_path == "":
+    print("please set AidaDB using --aidadb")
+    exit(1)
+if tmp_path == "":
+    tmp_path = tempfile.gettempdir()
+    print(f"tmp not set - using default {tmp_path}")
+
+# Block variables
+first_block = 0
+last_block = 60000000
+
+# --- Static variables --- #
+
+carmen_root = os.path.abspath('../go')
+
+# --- Script --- #
+
+# Log file path from which we read output to find kill_block
+aida_log_file = Path.cwd() / 'aida.log'
+carmen_log_file = Path.cwd() / 'carmen.log'
+current_dir = Path.cwd()
+
+print("Your settings:")
+print(f"\tNumber of iterations: {number_of_iterations}.")
+print(f"\tSync time before kill: {window} seconds.")
+print(f"\tCheckpoint granularity: {checkpoint_granularity} blocks.")
+
+
+# Function which checks programs return code, if program failed, log is printed and program is terminated.
+def check_program_failure(code, log):
+    if code != 0:
+        log.close()
+        with open(carmen_log_file, 'r') as l:
+            text = l.read()
+            print(text)
+        sys.exit(1)
+
+
+# Function which stops process after given sleep_time.
+def terminate_process_after(sleep_time: int, checkpoint: int):
+    start = 0.0
+    with aida_log_file.open('r') as f:
+        while True:
+            line = f.readline()
+            if not line:
+                time.sleep(0.1)
+                continue
+            if start > 0 and time.time() - start >= sleep_time:
+                print("Interrupting...")
+                process.terminate()
+                return checkpoint
+            if f"block {checkpoint + checkpoint_granularity}" in line:
+                # First checkpoint was found, we should start timer as it means block processing is running.
+                if start == 0.0:
+                    start = time.time()
+                checkpoint = checkpoint + checkpoint_granularity
+            # If process ends with error (return code 1) or either 'fail' or 'exit status' occurs in line exit script
+            if process.poll() == 1 or any(s in line for s in ["exit status", "fail"]):
+                print("Error occurred - printing output.log:")
+                with open(aida_log_file, 'r') as l:
+                    text = l.read()
+                    print(text)
+                return -1
+
+
+# First iteration command
+cmd = [
+    './build/aida-vm-sdb', 'substate', '--validate',
```

Comment:
as I mention previously, it would make more sense to me to configure the script directly to run `aida-vm-sdb` - why it has to be the Aida workspace as it appears in git?

Reply:
if the script requires Aida workspace, it would make more sense to run the command using `go run` so the user does not have to build Aida before

Reply:
As written in [this](https://github.com/Fantom-foundation/Carmen/pull/986#discussion_r1727097445) comment, using `go run` had issues...

Reply:
Discussed in DMs.